src/tests/homework1.spec.js

```diff
@@ -1,170 +1,142 @@
-    const emailInput = page.getByLabel('Email');
-    const passwordInput = page.getByLabel('heslo');
-    const controlPasswordInput = page.getByLabel('Kontrola hesla');
-    const registerButton = page.getByRole('button', { name: 'Zaregistrovat' });
-    const heading = page.locator('h1');
+    //test uses different fake names and email addresses for each run
+    test("valid registration 2", async ({ page }) => {
 
-    const randomName = faker.person.fullName();
-    await nameInput.fill(randomName);
+        const nameInput = page.getByLabel('Jméno a příjmení');
+        const emailInput = page.getByLabel('Email');
+        const registerButton = page.getByRole('button', { name: 'Zaregistrovat' });
+        const heading = page.locator('h1');
 
-    const randomEmail = faker.internet.email();
-    await emailInput.fill(randomEmail);
+        const randomName = faker.person.fullName();
+        await nameInput.fill(randomName);
 
-    await passwordInput.fill('noveHeslo1');
-    await controlPasswordInput.fill('noveHeslo1');
+        const randomEmail = faker.internet.email();
+        await emailInput.fill(randomEmail);
 
-    await registerButton.click();
-    const currentNewUserName = page
-        .locator('.navbar-right')
-        .locator('strong');
-    await expect(currentNewUserName).toHaveText(randomName);
-    await expect(heading).toHaveText('Přihlášky');
+        await fillPassword(page);
 
-});
+        await registerButton.click();
+        const currentNewUserName = page
+            .locator('.navbar-right')
+            .locator('strong');
+        await expect(currentNewUserName).toHaveText(randomName);
+        await expect(heading).toHaveText('Přihlášky');
 
-test("should not register with an existing email", async ({ page }) => {
+    });
 
-    //first registration + logout -> get email
-    await page.goto('/registrace');
+    test("should not register with an existing email", async ({ page }) => {
 
-    const nameInput = page.getByLabel('Jméno a příjmení');
-    const emailInput = page.getByLabel('Email');
-    const passwordInput = page.getByLabel('heslo');
-    const controlPasswordInput = page.getByLabel('Kontrola hesla');
-    const registerButton = page.getByRole('button', { name: 'Zaregistrovat' });
-    const logoutButton = page.locator('#logout-link');
+        //first registration + logout -> get certain email
+        const nameInput = page.getByLabel('Jméno a příjmení');
+        const emailInput = page.getByLabel('Email');
+        const registerButton = page.getByRole('button', { name: 'Zaregistrovat' });
+        const logoutButton = page.locator('#logout-link');
 
-    var randomName = faker.person.fullName();
-    await nameInput.fill(randomName);
+        var randomName = faker.person.fullName();
+        await nameInput.fill(randomName);
 
-    const randomEmail = faker.internet.email();
-    await emailInput.fill(randomEmail);
+        const randomEmail = faker.internet.email();
```

Comment:
faker je boží a velmi chválím, bylo by fajn si testovací data, které faker vygeneruje, zalogovat (třeba console.log). Jinak totiž nevíme na jakých datech přesně padl a zkomplikuje nám to práci.